app/grandchallenge/discussion_forums/models.py

```diff
@@ -230,6 +240,11 @@ def save(self, *args, **kwargs):
 
         if adding:
             self.assign_permissions()
+            # mark topic (and hence this post) as read by the user
+            TopicReadRecord.objects.update_or_create(
+                user=self.creator,
+                topic=self.topic,
+            )
```

Comment:
I'm wondering if we should do a permission check that `self.creator` has view permission for the topic. Where would be the best place for such a check? In the `save` method of the `TopicReadRecord` model? 
We're currently creating `TopicReadRecord`s here and in the view - in the view, we are already checking the permission.